### PR TITLE
[python3] Enable parallel builds; use vcpkg_install_msbuild()

### PR DESCRIPTION
**Pull request by ras0219**
See Title

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]

File: ports/python3/portfile.cmake [hunk 1]
```diff
@@ -53,42 +53,35 @@ if (VCPKG_TARGET_IS_WINDOWS)
 		message(FATAL_ERROR "Unsupported architecture: ${VCPKG_TARGET_ARCHITECTURE}")
 	endif()
 
-	vcpkg_build_msbuild(
-		PROJECT_PATH ${SOURCE_PATH}/PCBuild/pythoncore.vcxproj
-		PLATFORM ${BUILD_ARCH})
+	set(ENV{_CL_} "/FS /MP${VCPKG_CONCURRENCY}")
```
Comment (BillyONeal) [inline ports/python3/portfile.cmake:56]: I don't think we should be touching the `_CL_` escape hatch in a production scenario.

Comment (Voskrese) [reply ports/python3/portfile.cmake:56]: https://docs.microsoft.com/en-us/cpp/build/reference/cl-environment-variables?view=vs-2019

> this works, parallelizes without errors
> I made changes to the global settings for myself

`vcpkg_install_msbuild.cmake`

```
    endif()

    set(ENV{_CL_} "/FS /MP${VCPKG_CONCURRENCY}")

    list(APPEND _csc_OPTIONS
```

Comment (ras0219-msft) [reply ports/python3/portfile.cmake:56]: Given that we don't have a more reliable way to pass options into MSBuild (a patch would require modifying many, many project files), I think this is ok.

We should also probably consider moving this change (and improving it) in `vcpkg_install_msbuild` as @Voskrese suggests.

Comment (BillyONeal) [reply ports/python3/portfile.cmake:56]: >Given that we don't have a more reliable way to pass options into MSBuild (a patch would require modifying many, many project files), I think this is ok.

I would rather ignore `VCPKG_CONCURRENCY` than press the `_CL_` escape hatch into a production scenario.

Comment (ras0219-msft) [reply ports/python3/portfile.cmake]: I think I found a better way: `/p:CL_MPCount=${VCPKG_CONCURRENCY}`. The docs indicate that `/FS` should be automatically enabled, but I recall needing to explicitly specify it in my testing....

**Comment by strega-nil** [comment]
@ras0219 could you merge with master?

**Comment by ras0219** [comment]
After further work on this, it seems that `/p:CL_MPCount` is not effective.

However, I think that using the `CL` environment variable is appropriate for this specific case for two reasons:
1. It is common practice elsewhere to specify flags via environment variables (CXXFLAGS)
2. This flag is a CL-specific flag -- if some other compiler was to be used inside msbuild, it is desirable for it to be ignored. Therefore, the `CL` environment variable functions perfectly for this use.

While testing, I found two additional significant performance issues:
1. `file(COPY)` is ~3x slower than `xcopy` for duplicating the source tree.
2. Spawning ~20 `powershell.exe` processes takes a long time during `vcpkg_copy_tool_dependencies()`

After addressing each of these points, the reported time for `vcpkg build python3:x64-windows` on my machine is down to 38.25s from 1.331mins -- ~2x faster.

**Comment by BillyONeal** [comment]
> However, I think that using the `CL` environment variable is appropriate for this specific case for two reasons:
> 
> 1. It is common practice elsewhere to specify flags via environment variables (CXXFLAGS)

I would object much less to `CL` rather than `_CL_`. The former I've seen used in build systems sometimes, particularly make derived ones. The latter is the escape hatch for the former.

File: scripts/buildsystems/msbuild/applocal.ps1 [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
-param([string]$targetBinary, [string]$installedDir, [string]$tlogFile, [string]$copiedFilesLog)
+param([string[]]$targetBinary, [string[]]$installedDir, [string]$tlogFile, [string]$copiedFilesLog)
 
+$installedDirs = $installedDir
+$targetBinaries = $targetBinary
+
+$dumpbinCache = @{}
+
+foreach ($targetBinary in $targetBinaries) {
+foreach ($installedDir in $installedDirs) {
```
Comment (strega-nil) [inline scripts/buildsystems/msbuild/applocal.ps1:10]: Why aren't these indented?

**Comment by ras0219** [comment]
This is blocked on fixes in #13019 and #13126

**Pull request closed without merging**